File: bin/com/problem/go/googleKickstart/2022/roundg/FindCandy3.go
package main

import (
  "fmt"
)

func solve(N int, M int) {
  /**
  * 1. sum all the shit
  * 2. divide and get the remaining
  */
  var total int
  for i := 0; i < N; i++ {
    var S int
    fmt.Scan(&S)
    total += S
  }

  fmt.Println(total % M)
}

func main() {
  var T int 
  fmt.Scan(&T)
  for t := 1; t <= T; t++ {
    var N int
    var M int
    fmt.Scanf("%d %d", &N, &M)
    fmt.Printf("Case #%d: ", t)
    solve(N, M)
  }
}

File: com/problem/java/googleKickstart/2022/session/3/FindCandy3.java
import java.util.*;
import java.io.*;

public class Solution {
  static Scanner sc = new Scanner(System.in);
  static PrintStream out = System.out;
  static void solve(int numOfDays, int totalKids) {
    /**
     * constant time not using array or arraylist
     * 1. sum all of the bags of candy
     * 2. divide and find the remaining
     **/

    int total = 0;
    for (int i = 0; i < numOfDays; ++i) {
      total += sc.nextInt();
    }

    out.println(total % totalKids);
  }

  public static void main(String[] args) {
    int T = sc.nextInt();

    for (int t = 1; t <= T; ++t) {
      int N = sc.nextInt();
      int M = sc.nextInt();
      out.print("Case #"+t+": ");
      solve(N, M);
    }
    sc.close();
  }
}
